src/Ecs/Systems/PlayerSwitchSystem.gd

```diff
@@ -8,19 +8,27 @@ var _event_bus
 func _init(event_bus):
 	self.required_component_types = ["PlayableComponent"]
 	_event_bus = event_bus
+	_event_bus.connect("activate_tile", self, "on_activate_tile")
 
 func _process(delta):
 	if Input.is_action_just_pressed("switch_player"):
-		_switch_player()
+		_switch_player_to()
 
-func _switch_player():
+func on_activate_tile(tile):
+	for e in entities:
+		if e.position == tile and not e.has("PlayerControlComponent"):
+			_switch_player_to(e)
+
+func _switch_player_to(to=null):
 	var from
 	for e in entities:
 		if e.has("PlayerControlComponent"):
 			from = e
 			break
 	
-	var to = entities[entities.find(from) -1]
+	if to == null:
+		to = entities[entities.find(from) -1]
```

Comment:
I don't understand what this is doing. When is `to` null and what does this do?

Reply:
If the entity to switch to isn't specified(when using the hotkey) it picks a playable entity with some index magic.

The entity to switch to is specified when switching by tapping on it.